content/posts/2015-09-02-heroku-build-static.md

```diff
@@ -0,0 +1,119 @@
+title: Node.js static file build steps in Python Heroku apps
+date: 2015-09-02
+category: posts
+tags: [heroku, python, node]
+---
+
+Python webapps are pretty cool. Most frontend tools end up getting written in
+Node.js. Things like Less for nicer style sheets, Babel to write modern JS
+today, or NPM to manage front-end dependencies. The result is a lot of webapps
+are polyglots that require more than just Python or Node.
+
+In most workflows, this seems to be largely figured out. Just run all the tools.
+Most READMEs I've seen lately have a box something like this
+
+```bash
+$ git clone https://github.example.com/foo/bar.git
+$ cd git
+$ pip install -r requirements.txt
+$ npm install
+$ gulp static-assets
+$ python ./manage.py runserver
+```
+
+Unfortunately, Heroku doesn't seem to have gotten this memo. There is a Python
+buildpack, and a Node buildpack, but no clear way of combining the two.
+
+Multi Buildpack
+===============
+
+[GitHub is littered with attempts to fix this by building new buildpacks.][search]
+The problem with the versions of this I looked at is they invariable fall out of
+compatibility with Heroku. I could probably fix them up to work, but then *I'd*
+have to maintain them. Not maintaining infrastructure like this is the reason I
+use Heroku, so this seems like one step forward, one step back.
```

Comment:
In which we kill unnecessary words with prejudice.

> The problem is that custom buildpacks invariably fall out of compatibility with Heroku.

> I could probably fix them, but...

> I use Heroku to avoid maintaining infrastructure; custom buildpacks are one step forward, but two steps back.